geonode/utils.py

```diff
@@ -2121,3 +2121,10 @@ def verify_image(stream):
     except Exception as e:
         logger.debug(e)
         return False
+
+
+def find_by_attr(lst, val, attr="id"):
+    """ Returns an object if the id matches in any list of objects """
+    for item in lst:
+        if item[attr] == val:
```

Comment:
Did you evaluate using the `.get` method of the dictionary? Because in this case if `attr` does not exist in the` item` dict, it will raise an exception.
Maybe something like can avoid it:

```python
def find_by_attr(lst, val, attr="id"):
    """ Returns an object if the id matches in any list of objects """
    for item in lst:
        if item.get(attr, None):
            return item
```

Reply:
We can put a check on that to return None in that case. Actually it was a last minute change before PR commit when I found the same code is repeated in both tests and model. It was initially only for id only in tests. I will put a check.